sdk/python/feast/on_demand_feature_view.py

```diff
@@ -631,7 +631,7 @@ def mainify(obj) -> None:
             obj.__module__ = "__main__"
 
     def decorator(user_function):
-        return_annotation = inspect.signature(user_function).return_annotation
+        return_annotation = get_type_hints(user_function).get("return", inspect._empty)
```

Comment:
Do we need to default to `inspect._empty` anymore after the change? Maybe just default to `None` and change if conditions accordingly?

Reply:
None conflicts with return type None. So using inspect._empty when the return type is missing.